File: autohotkey.py
class AutoHotkeyLexer(RegexLexer):
    """
    For AutoHotkey source code with preprocessor directives.
    """
    name = 'autohotkey'
    aliases = ['ahk']
    filenames = ['*.ahk', '*.ahkl']
    flags = re.IGNORECASE
    #: optional Comment or Whitespace
    _ws = r'(?:\s|;.*?\n|/[*].*?[*]/)+'

    tokens = {
        'whitespace': [
            (r'^\s*#if\s+0', Comment.Preproc, 'if0'),
            (r'^\s*#', Comment.Preproc, 'macro'),
            (r'\n', Text),
            (r'\s+', Text),
             (r';(\n|(.|\n)*?[^\\]\n)', Comment),
            (r'/(\\\n)?[*](.|\n)*?[*](\\\n)?/', Comment),
        ],
        'statements': [
            (r'L?"', String, 'string'),
            (r"L?'(\\.|\\[0-7]{1,3}|\\x[a-fA-F0-9]{1,2}|[^\\\'\n])'", String.Char),
            (r'(\d+\.\d*|\.\d+|\d+)[eE][+-]?\d+[lL]?', Number.Float),
            (r'(\d+\.\d*|\.\d+|\d+[fF])[fF]?', Number.Float),
            (r'0x[0-9a-fA-F]+[Ll]?', Number.Hex),
            (r'\d+[Ll]?', Number.Integer),
            (r'[~!%^&*+=|?:<>/-]', Operator),
            (r'[()\[\],.]', Punctuation),
            (r'\b(case)(.+?)(:)', bygroups(Keyword, using(this), Text)),
            (r'(auto|break|case|const|continue|default|do|else|enum|extern|'
             r'%.*%'
             r'RTrim|JoinLow|BelowNormal|Normal|AboveNormal|'
             r'High|Realtimeahk_id|ahk_pid|ahk_class|ahk_groupBetween|'
             r'Contains|In|Is|Integer|Float|'
             r'IntegerFast|FloatFast|Number|Digit|Xdigit|'
             r'Alpha|Upper|Lower|Alnum|Time|'
             r'DateNot|Or|AndAlwaysOnTop|Topmost|Top|'
             r'Bottom|Transparent|TransColor|Redraw|Region|'
             r'ID|IDLast|ProcessName|MinMax|ControlList|'
             r'Count|List|Capacity|StatusCD|Eject|'
             r'Lock|Unlock|Label|FileSystem|Label|'
             r'SetLabel|Serial|Type|Statusstatic|global|'
             r'local|ByRefSeconds|Minutes|Hours|DaysRead|'
             r'ParseLogoff|Close|Error|SingleTray|Add|'
             r'Rename|Check|UnCheck|ToggleCheck|Enable|'
             r'Disable|ToggleEnable|Default|NoDefault|Standard|'
             r'NoStandard|Color|Delete|DeleteAll|Icon|'
             r'NoIcon|Tip|Click|Show|MainWindow|'
             r'NoMainWindow|UseErrorLevelText|Picture|Pic|GroupBox|'
             r'Button|Checkbox|Radio|DropDownList|DDL|'
             r'ComboBox|ListBox|ListView|DateTime|MonthCal|'
             r'Slider|StatusBar|Tab|Tab2|TreeView|'
             r'UpDownIconSmall|Tile|Report|SortDesc|NoSort|'
             r'NoSortHdr|Grid|Hdr|AutoSize|Rangexm|'
             r'ym|ys|xs|xp|ypFont|'
             r'Resize|Owner|Submit|NoHide|Minimize|'
             r'Maximize|Restore|NoActivate|NA|Cancel|'
             r'Destroy|CenterMargin|MaxSize|MinSize|OwnDialogs|'
             r'GuiEscape|GuiClose|GuiSize|GuiContextMenu|GuiDropFilesTabStop|'
             r'Section|AltSubmit|Wrap|HScroll|VScroll|'
             r'Border|Top|Bottom|Buttons|Expand|'
             r'First|ImageList|Lines|WantCtrlA|WantF2|'
             r'Vis|VisFirst|Number|Uppercase|Lowercase|'
             r'Limit|Password|Multi|WantReturn|Group|'
             r'Background|bold|italic|strike|underline|'
             r'norm|BackgroundTrans|Theme|Caption|Delimiter|'
             r'MinimizeBox|MaximizeBox|SysMenu|ToolWindow|Flash|'
             r'Style|ExStyle|Check3|Checked|CheckedGray|'
             r'ReadOnly|Password|Hidden|Left|Right|'
             r'Center|NoTab|Section|Move|Focus|'
             r'Hide|Choose|ChooseString|Text|Pos|'
             r'Enabled|Disabled|Visible|LastFound|LastFoundExistAltTab|'
             r'ShiftAltTab|AltTabMenu|AltTabAndMenu|AltTabMenuDismissNoTimers|Interrupt|'
             r'Priority|WaitClose|Wait|Exist|Close{Blind}|'
             r'Mod|Pow|Exp|Sqrt|Log|'
             r'Ln|Round|Ceil|Floor|Abs|'
             r'Sin|Cos|Tan|ASin|ACos|'
             r'ATan|BitNot|BitAnd|BitOr|BitXOr|'
             r'BitShiftLeft|BitShiftRightYes|No|Ok|Cancel|'
             r'Abort|Retry|Ignore|TryAgainOn|Off|'
             r'for|goto|if|register|restricted|return|sizeof|static|struct|'
             r'switch|typedef|union|volatile|virtual|while)\b', Keyword),
            (r'(global|static|local)\b',
             Keyword.Type),
            (r'(_{0,2}inline|naked|restrict|thread|typename)\b', Keyword.Reserved),
            (r'__(asm|int8|based|except|int16|stdcall|cdecl|fastcall|int32|'
             r'true|false|NULL|try|leave)\b', Keyword.Reserved),
            (r'(true|false|NULL)\b'
             r'BlockInput|Break|Click|ClipWait|Continue|'
             r'Control|ControlClick|ControlFocus|ControlGet|ControlGetFocus|'
             r'ControlGetPos|ControlGetText|ControlMove|ControlSend|ControlSendRaw|'
             r'ControlSetText|CoordMode|Critical|DetectHiddenText|DetectHiddenWindows|'
             r'Drive|DriveGet|DriveSpaceFree|Edit|Else|'
             r'EnvAdd|EnvDiv|EnvGet|EnvMult|EnvSet|'
             r'EnvSub|EnvUpdate|Exit|ExitApp|FileAppend|'
             r'FileCopy|FileCopyDir|FileCreateDir|FileCreateShortcut|FileDelete|'
             r'FileGetAttrib|FileGetShortcut|FileGetSize|FileGetTime|FileGetVersion|'
             r'FileInstall|FileMove|FileMoveDir|FileRead|FileReadLine|'
             r'FileRecycle|FileRecycleEmpty|FileRemoveDir|FileSelectFile|FileSelectFolder|'
             r'FileSetAttrib|FileSetTime|FormatTime|GetKeyState|Gosub|'
             r'Goto|GroupActivate|GroupAdd|GroupClose|GroupDeactivate|'
             r'Gui|GuiControl|GuiControlGet|Hotkey|If Else|'
             r'IfExist|IfGreater|IfGreaterOrEqual|IfInString|IfLess|'
             r'IfLessOrEqual|IfMsgBox|IfNotEqual|IfNotExist|IfNotInString|'
             r'IfWinActive|IfWinExist|IfWinNotActive|IfWinNotExist|ImageSearch|'
             r'IniDelete|IniRead|IniWrite|Input|InputBox|'
             r'KeyHistory|KeyWait|ListHotkeys|ListLines|ListVars|'
             r'Loop|Loop, FilePattern|Loop, Parse|Loop, Read|Loop, Reg|'
             r'Menu|MouseClick|MouseClickDrag|MouseGetPos|MouseMove|'
             r'MsgBox|OnExit|OutputDebug|Pause|PixelGetColor|OnMessage|'
             r'PixelSearch|PostMessage|Process|Progress|Random|'
             r'RegExMatch|RegExReplace|IsFunc|DllCall|'
             r'RegDelete|RegRead|RegWrite|Reload|Repeat|'
             r'Return|Run|RunAs|RunWait|Send|'
             r'SendEvent|SendInput|SendMessage|SendMode|SendPlay|'
             r'SendRaw|SetBatchLines|SetCapslockState|SetControlDelay|SetDefaultMouseSpeed|'
             r'SetEnv|SetFormat|SetKeyDelay|SetMouseDelay|SetNumlockState|'
             r'SetScrollLockState|SetStoreCapslockMode|SetTimer|SetTitleMatchMode|SetWinDelay|'
             r'SetWorkingDir|Shutdown|Sleep|Sort|SoundBeep|'
             r'SoundGet|SoundGetWaveVolume|SoundPlay|SoundSet|SoundSetWaveVolume|'
             r'SplashImage|SplashTextOff|SplashTextOn|SplitPath|StatusBarGetText|'
             r'StatusBarWait|StringCaseSense|StringGetPos|StringLeft|StringLen|'
             r'StringLower|StringMid|StringReplace|StringRight|StringSplit|'
             r'StringTrimLeft|StringTrimRight|StringUpper|Suspend|SysGet|'
             r'Thread|ToolTip|Transform|TrayTip|URLDownloadToFile|'
             r'While|WinActivate|WinActivateBottom|WinClose|WinGet|'
             r'WinGetActiveStats|WinGetActiveTitle|WinGetClass|WinGetPos|WinGetText|'
             r'WinGetTitle|WinHide|WinKill|WinMaximize|WinMenuSelectItem|'
             r'WinMinimize|WinMinimizeAll|WinMinimizeAllUndo|WinMove|WinRestore|'
             r'WinSet|WinSetTitle|WinShow|WinWait|WinWaitActive|'
             r'WinWaitClose|WinWaitNotActive', Name.Builtin),
            ('[a-zA-Z_][a-zA-Z0-9_]*:(?!:)', Name.Label),
            (r'[a-zA-Z_][a-zA-Z0-9_]*::(?!:)', Name.Label),
            (r'::[a-zA-Z_][a-zA-Z0-9_]*::(?!:)', Name.Label),
            ('[a-zA-Z_][a-zA-Z0-9_]*', Name),
        ],
        'root': [
            include('whitespace'),
            # functions
            (r'((?:[a-zA-Z0-9_*\s])+?(?:\s|[*]))'    # return arguments
             r'([a-zA-Z_][a-zA-Z0-9_]*)'             # method name
             r'(\s*\([^;]*?\))'                      # signature
             r'(' + _ws + r')({)',
             bygroups(using(this), Name.Function, using(this), using(this),
                      Punctuation),
             'function'),
            # function declarations
            (r'((?:[a-zA-Z0-9_*\s])+?(?:\s|[*]))'    # return arguments
             r'([a-zA-Z_][a-zA-Z0-9_]*)'             # method name
             r'(\s*\([^;]*?\))'                      # signature
             r'(' + _ws + r')(;)',
             bygroups(using(this), Name.Function, using(this), using(this),
                      Punctuation)),
            ('', Text, 'statement'),
        ],
        'statement' : [
            include('whitespace'),
            include('statements'),
            ('[{}]', Punctuation),
            (';', Punctuation, '#pop'),
        ],
        'function': [
            include('whitespace'),
            include('statements'),
            ('{', Punctuation, '#push'),
            ('}', Punctuation, '#pop'),
        ],
        'string': [
            (r'"', String, '#pop'),
            (r'\\([\\abfnrtv"\']|x[a-fA-F0-9]{2,4}|[0-7]{1,3})', String.Escape),
            (r'[^`"\n]+', String), # all other characters
            (r'\\', String), # stray backslash
        ],
        'macro': [
            (r'[^/\n]+', Comment.Preproc),
            (r'/[*](.|\n)*?[*]/', Comment),
            (r'//.*?\n', Comment, '#pop'),
            (r'/', Comment.Preproc),
            (r'(?<=\\)\n', Comment.Preproc),
            (r'\n', Comment.Preproc, '#pop'),
        ],
        'if0': [
            (r'^\s*#if.*?(?<!\\)\n', Comment, '#push'),
            (r'^\s*#el(?:se|if).*\n', Comment.Preproc, '#pop'),
            (r'^\s*#endif.*?(?<!\\)\n', Comment, '#pop'),
            (r'.*?\n', Comment),
        ]
    }

    stdlib_types = ['global', 'local', 'static']

    def __init__(self, **options):
        self.stdlibhighlighting = get_bool_opt(options,
                'stdlibhighlighting', True)
        self.c99highlighting = get_bool_opt(options,
                'c99highlighting', True)
        RegexLexer.__init__(self, **options)

    def get_tokens_unprocessed(self, text):
        for index, token, value in \
            RegexLexer.get_tokens_unprocessed(self, text):
            if token is Name:
                if self.stdlibhighlighting and value in self.stdlib_types:
                    token = Keyword.Type
            yield index, token, value

File: autohotkey.php
<?php
/*************************************************************************************
 * autohotkey.php
 * --------
 * Author: naveen garg (naveen.garg@gmail.com)
 * Copyright: (c) 2009 and to GESHi ;)
 * Release Version: 0.2
 * Date Started: 2009/06/11
 *
 * Autohotkey language file for GeSHi.
 *
 * CHANGES
 * -------
 * Release 0.1 (2009/06/11)
 * - First Release
 *
 * Current bugs & todo:
 * ----------
 * Reference: http://www.autohotkey.com/docs/
 *************************************************************************************
 *
 *    This file is part of GeSHi.
 *
 *   GeSHi is free software; you can redistribute it and/or modify
 *   it under the terms of the GNU General Public License as published by
 *   the Free Software Foundation; either version 2 of the License,
or
 *   (at your option) any later version.
 *
 *   GeSHi is distributed in the hope that it will be useful,
 *   but WITHOUT ANY WARRANTY; without even the implied warranty of
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *   GNU General Public License for more details.
 *
 *   You should have received a copy of the GNU General Public License
 *   along with GeSHi; if not,
write to the Free Software
 *   Foundation,
Inc.,
59 Temple Place,
Suite 330,
Boston,
MA  02111-1307  USA
 *
 ************************************************************************************/

$language_data = array (
    'LANG_NAME' => 'Autohotkey',
    'COMMENT_SINGLE' => array(';'),
    'COMMENT_MULTI' => array('/*' => '*/'),
    'CASE_KEYWORDS' => GESHI_CAPS_NO_CHANGE,
    'QUOTEMARKS' => array('"'),
    'ESCAPE_CHAR' => '',
    'KEYWORDS' => array(
    1 => array(
    'while','if','and','or','else','return'
    ),
    2 => array('A_(\w+)'  // built in variables
    ),
    3 => array(
    'AutoTrim',
    'BlockInput','Break','Click',
    'ClipWait','Continue','Control',
    'ControlClick','ControlFocus','ControlGet',
    'ControlGetFocus','ControlGetPos','ControlGetText',
    'ControlMove','ControlSend','ControlSendRaw',
    'ControlSetText','CoordMode','Critical',
    'DetectHiddenText','DetectHiddenWindows','DllCall','Drive',
    'DriveGet','DriveSpaceFree',
    'Else','EnvAdd','EnvDiv',
    'EnvGet','EnvMult','EnvSet',
    'EnvSub','EnvUpdate','Exit',
    'ExitApp','FileAppend','FileCopy',
    'FileCopyDir','FileCreateDir','FileCreateShortcut',
    'FileDelete','FileGetAttrib','FileGetShortcut',
    'FileGetSize','FileGetTime','FileGetVersion',
    'FileInstall','FileMove','FileMoveDir',
    'FileRead','FileReadLine','FileRecycle',
    'FileRecycleEmpty','FileRemoveDir','FileSelectFile',
    'FileSelectFolder','FileSetAttrib','FileSetTime',
    'FormatTime','Gosub',
    'Goto','GroupActivate','GroupAdd',
    'GroupClose','GroupDeactivate','Gui',
    'GuiControl','GuiControlGet','Hotkey',
    'IfExist','IfGreater','IfGreaterOrEqual',
    'IfInString','IfLess','IfLessOrEqual',
    'IfMsgBox','IfNotEqual','IfNotExist',
    'IfNotInString','IfWinActive','IfWinExist',
    'IfWinNotActive','IfWinNotExist','ImageSearch',
    'IniDelete','IniRead','IniWrite',
    'Input','InputBox','KeyHistory',
    'KeyWait','ListHotkeys','ListLines',
    'ListVars','Loop',
    'Menu','MouseClick','MouseClickDrag',
    'MouseGetPos','MouseMove','MsgBox',
    'OnMessage','OnExit','OutputDebug',
    'PixelGetColor','PixelSearch','PostMessage',
    'Process','Progress','Random',
    'RegExMatch','RegExReplace','RegisterCallback',
    'RegDelete','RegRead','RegWrite',
    'Reload','Repeat','Return',
    'Run','RunAs','RunWait',
    'Send','SendEvent','SendInput',
    'SendMessage','SendMode','SendPlay',
    'SendRaw','SetBatchLines','SetCapslockState',
    'SetControlDelay','SetDefaultMouseSpeed','SetEnv',
    'SetFormat','SetKeyDelay','SetMouseDelay',
    'SetNumlockState','SetScrollLockState','SetStoreCapslockMode',
    'SetTimer','SetTitleMatchMode','SetWinDelay',
    'SetWorkingDir','Shutdown','Sleep',
    'Sort','SoundBeep','SoundGet',
    'SoundGetWaveVolume','SoundPlay','SoundSet',
    'SoundSetWaveVolume','SplashImage','SplashTextOff',
    'SplashTextOn','SplitPath','StatusBarGetText',
    'StatusBarWait','StringCaseSense','StringGetPos',
    'StringLeft','StringLen','StringLower',
    'StringMid','StringReplace','StringRight',
    'StringSplit','StringTrimLeft','StringTrimRight',
    'StringUpper','Suspend','SysGet',
    'Thread','ToolTip','Transform',
    'TrayTip','URLDownloadToFile','While',
    'VarSetCapacity',
    'WinActivate','WinActivateBottom','WinClose',
    'WinGet','WinGetActiveStats','WinGetActiveTitle',
    'WinGetClass','WinGetPos','WinGetText',
    'WinGetTitle','WinHide','WinKill',
    'WinMaximize','WinMenuSelectItem','WinMinimize',
    'WinMinimizeAll','WinMinimizeAllUndo','WinMove',
    'WinRestore','WinSet','WinSetTitle',
    'WinShow','WinWait','WinWaitActive',
    'WinWaitClose','WinWaitNotActive'
    ),
    4 => array(
    'Abs','ACos','Asc','ASin',
    'ATan','Ceil','Chr','Cos',
    'Exp','FileExist','Floor',
    'GetKeyState','IL_Add','IL_Create','IL_Destroy',
    'InStr','IsFunc','IsLabel','Ln',
    'Log','LV_Add','LV_Delete','LV_DeleteCol',
    'LV_GetCount','LV_GetNext','LV_GetText','LV_Insert',
    'LV_InsertCol','LV_Modify','LV_ModifyCol','LV_SetImageList',
    'Mod','NumGet','NumPut',
    'Round',
    'SB_SetIcon','SB_SetParts','SB_SetText','Sin',
    'Sqrt','StrLen','SubStr','Tan',
    'TV_Add','TV_Delete','TV_GetChild','TV_GetCount',
    'TV_GetNext','TV_Get','TV_GetParent','TV_GetPrev',
    'TV_GetSelection','TV_GetText','TV_Modify',
    'WinActive','WinExist'
    ),
    5 => array(   // #Directives
    'AllowSameLineComments','ClipboardTimeout','CommentFlag',
    'ErrorStdOut','EscapeChar','HotkeyInterval',
    'HotkeyModifierTimeout','Hotstring','IfWinActive',
    'IfWinExist','IfWinNotActive','IfWinNotExist',
    'Include','IncludeAgain','InstallKeybdHook',
    'InstallMouseHook','KeyHistory','LTrim',
    'MaxHotkeysPerInterval','MaxMem','MaxThreads',
    'MaxThreadsBuffer','MaxThreadsPerHotkey','NoEnv',
    'NoTrayIcon','Persistent','SingleInstance',
    'UseHook','WinActivateForce'
    ),
    6 => array(
    'Shift','LShift','RShift',
    'Alt','LAlt','RAlt',
    'LControl','RControl',
    'Ctrl','LCtrl','RCtrl',
    'LWin','RWin','AppsKey',
    'AltDown','AltUp','ShiftDown',
    'ShiftUp','CtrlDown','CtrlUp',
    'LWinDown','LWinUp','RWinDown',
    'RWinUp','LButton','RButton',
    'MButton','WheelUp','WheelDown',
    'WheelLeft','WheelRight','XButton1',
    'XButton2','Joy1','Joy2',
    'Joy3','Joy4','Joy5',
    'Joy6','Joy7','Joy8',
    'Joy9','Joy10','Joy11',
    'Joy12','Joy13','Joy14',
    'Joy15','Joy16','Joy17',
    'Joy18','Joy19','Joy20',
    'Joy21','Joy22','Joy23',
    'Joy24','Joy25','Joy26',
    'Joy27','Joy28','Joy29',
    'Joy30','Joy31','Joy32',
    'JoyX','JoyY','JoyZ',
    'JoyR','JoyU','JoyV',
    'JoyPOV','JoyName','JoyButtons',
    'JoyAxes','JoyInfo','Space',
    'Tab','Enter',
    'Escape','Esc','BackSpace',
    'BS','Delete','Del',
    'Insert','Ins','PGUP',
    'PGDN','Home','End',
    'Up','Down','Left',
    'Right','PrintScreen','CtrlBreak',
    'Pause','ScrollLock','CapsLock',
    'NumLock','Numpad0','Numpad1',
    'Numpad2','Numpad3','Numpad4',
    'Numpad5','Numpad6','Numpad7',
    'Numpad8','Numpad9','NumpadMult',
    'NumpadAdd','NumpadSub','NumpadDiv',
    'NumpadDot','NumpadDel','NumpadIns',
    'NumpadClear','NumpadUp','NumpadDown',
    'NumpadLeft','NumpadRight','NumpadHome',
    'NumpadEnd','NumpadPgup','NumpadPgdn',
    'NumpadEnter','F1','F2',
    'F3','F4','F5',
    'F6','F7','F8',
    'F9','F10','F11',
    'F12','F13','F14',
    'F15','F16','F17',
    'F18','F19','F20',
    'F21','F22','F23',
    'F24','Browser_Back','Browser_Forward',
    'Browser_Refresh','Browser_Stop','Browser_Search',
    'Browser_Favorites','Browser_Home','Volume_Mute',
    'Volume_Down','Volume_Up','Media_Next',
    'Media_Prev','Media_Stop','Media_Play_Pause',
    'Launch_Mail','Launch_Media','Launch_App1',
    'Launch_App2'  
    ),
    7 => array('Add',  // Gui commands  
    'Show', 'Submit', 'Cancel', 'Destroy',
    'Font', 'Color', 'Margin', 'Flash', 'Default', 
    'GuiEscape','GuiClose','GuiSize','GuiContextMenu','GuiDropFilesTabStop',
    ),
    8 => array('Button',  // Gui Controls
    'Checkbox','Radio','DropDownList','DDL',
    'ComboBox','ListBox','ListView',
    'Text', 'Edit', 'UpDown', 'Picture',
    'TreeView','DateTime', 'MonthCal',
    'Slider'
    ),

    ),
    'SYMBOLS' => array(
    '(',')','[',']',
    '+','-','*','/','&','^',
    '=','+=','-=','*=','/=','&=',
    '==','<','<=','>','>=',':=',
    ',','.'
    ),
    'CASE_SENSITIVE' => array(
    GESHI_COMMENTS => false,
    1 => false,
    2 => false,
    3 => false,
    4 => false,
    5 => false,
    6 => false,
    7 => false,
    8 => false
    ),
    'STYLES' => array(
    'KEYWORDS' => array(
    1 => 'color: #AAAAFF; font-weight: bold;',       // reserved #blue
    2 => 'color: #88FF88;',                         // BIV yellow
    3 => 'color: #FF00FF; font-style: italic;',       // commands purple
    4 => 'color: #888844; font-weight: bold;',       // functions #0080FF
    5 => 'color: #000000; font-style: italic;',    // directives #black
    6 => 'color: #FF0000; font-style: italic;',      // hotkeys #red
    7 => 'color: #000000; font-style: italic;',    // gui commands #black
    8 => 'color: #000000; font-style: italic;'      // gui controls
    ),
    'COMMENTS' => array(
    0 => 'font-style: italic; color: #009933;',
    'MULTI' => 'font-style: italic; color: #669900;'
    ),
    'ESCAPE_CHAR' => array(
    0 => ''
    ),
    'BRACKETS' => array(
    0 => 'color: #00FF00; font-weight: bold;'
    ),
    'STRINGS' => array(
    0 => 'font-weight: bold; color: #008080;'
    ),
    'NUMBERS' => array(
    0 => 'color: #0000dd;'
    ),
    'METHODS' => array(
    1 => 'color: #0000FF; font-style: italic; font-weight: italic;'
    ),
    'SYMBOLS' => array(
    0 => 'color: #000000; font-weight: italic;'
    ),
    'REGEXPS' => array(
    0 => 'font-weight: italic; color: #A00A0;',
    1 => 'color: #CC0000; font-style: italic;',
    2 => 'color: #DD0000; font-style: italic;'
    ),
    'SCRIPT' => array(
    )
    ),
    'OOLANG' => false,
    'OBJECT_SPLITTERS' => array(
    1 => '_'
    ),
    'REGEXPS' => array(
    //Variables
    0 => '%[a-zA-Z_][a-zA-Z0-9_]*%',
    //hotstrings
    1 => '::[\w\d]+::',
    //labels
    2 => '\w[\w\d]+:\s'  // labels
    ),    
    'URLS' => array(
    1 => '',
    2 => 'http://www.autohotkey.com/docs/Variables.htm#{FNAME}',
    3 => 'http://www.autohotkey.com/docs/commands/{FNAME}.htm',
    4 => 'http://www.autohotkey.com/docs/Functions.htm#BuiltIn',
    5 => 'http://www.autohotkey.com/docs/commands/_{FNAME}.htm',
    6 => '',
    7 => 'http://www.autohotkey.com/docs/commands/Gui.htm#{FNAME}',
    8 => 'http://www.autohotkey.com/docs/commands/GuiControls.htm#{FNAME}'
    ),
    'STRICT_MODE_APPLIES' => GESHI_MAYBE,
    'SCRIPT_DELIMITERS' => array(
    ),
    'HIGHLIGHT_STRICT_BLOCK' => array(
    0 => true,
    1 => true,
    2 => true,
    3 => true
    ),
    'PARSER_CONTROL' => array(
    'KEYWORDS' => array(
    5 => array(
    'DISALLOWED_BEFORE' => '(?<!\w)\#'
    )
    )
    )
);

?>